docs/content/en/docs/pipeline-stages/builders/_index.md

```diff
@@ -41,24 +41,37 @@ Skaffold supports building artifacts in following execution contexts:
 
 ## Local Build
 Local build execution is the default execution context.
-Skaffold will use the build tools locally installed on your machine to execute the build.
+Skaffold will use the build tools (such as Docker, Bazel, Maven or Gradle) locally installed on your machine to execute the build.
 
 **Configuration**
 
 To configure the local execution explicitly, add build type `local` to the build section of `skaffold.yaml`
 
 ```yaml
 build:
-  local:
-    ...
+  local: {}
 ```
 
+The following options can optionally be configured:
+
 {{< schema root="LocalBuild" >}}
 
-If you are deploying to [local cluster]({{<relref "/docs/environment/local-cluster" >}}), you can additional set `push` to `false` to speed up builds.
+**Faster builds**
+
+If you are deploying to [local cluster]({{<relref "/docs/environment/local-cluster" >}}), 
+Skaffold will default `push` to `false` to speed up builds.
```

Comment:
I wonder if we should expand on this paragraph because there are a few implications.  
  - certain clusters, like minikube, kind, Docker Kubernetes, are considered as _local clusters_
  - Skaffold can usually load the images directly into a local cluster, instead of via an image registry
  - local clusters change the default for push to `false` to avoid pushing the image to the remote repository
  - this means that the same builds can fail when switching the kube context to a remote cluster.  but the speed tradeoff is worth it.

And maybe _Faster builds_ isn't the right section heading either.

Reply:
I'm happy to add some more documentation when most of this is merged :-)